packages/ppx_version/ppx_version.0.1/opam

```diff
@@ -0,0 +1,18 @@
+opam-version: "1.2"
+name: "ppx_version"
+maintainer: "opensource@o1labs.org"
+authors: ["O(1) Labs, LLC <opensource@o1labs.org>"]
+homepage: "https://github.com/o1-labs/ppx_version"
+bug-reports: "https://github.com/o1-labs/ppx_version/issues"
+dev-repo: "git+https://github.com/o1-labs/ppx_version.git"
+license: "MIT"
+build: [
```

Comment:
I think you'll want to add the dependencies required by the library [here](https://github.com/o1-labs/ppx_version/blob/74df914452b6ea324020fd4a8780ce196f1bf14a/src/dune#L5). (I think `compiler_libs` there is automatically installed when installing the compiler, so don't worry about that one.)

Reply:
OK I see, I'm trying to figure out why dependencies are declared in so many places. I think I have an idea now:

* opam-repository `opam`: ?
* local `opam` or `package.opam`: will allow opam to install the missing dependencies when using `opam install` or `opam install --deps-only`
* `dune`: enforces that dependencies are present in workspace or installed when you build with `dune build`

It also looks like opam files can list build dependencies, so now I'm wondering if ppx libraries should be listed like that? I'm wondering if you don't have a ppx library installed, will `opam install --deps-only` figure out how to install it if's not listed a dependency (and only listed in the preprocessor section):

> Build depends: you may add, among others, the key build in the version constraints, e.g. "package" {build & >= "3.2"}, to indicate that there is no run-time dependency to this package: it is required but won't trigger rebuilds of your package when changed.